contracts/BytesUtil.sol

```diff
@@ -0,0 +1,62 @@
+pragma solidity ^0.4.24;
+
+library BytesUtil{
+    uint internal constant BYTES_HEADER_SIZE = 32;
+    uint internal constant WORD_SIZE = 32;
+    
+    // Returns a memory pointer to the data portion of the provided bytes array.
+    function dataPtr(bytes memory bts) internal pure returns (uint256 addr) {
+        assembly {
+            addr := add(bts, /*BYTES_HEADER_SIZE*/ 32)
+        }
+    }
+    
+    // Copy 'len' bytes from memory address 'src', to address 'dest'.
+    // This function does not check the or destination, it only copies
+    // the bytes.
+    function copy(uint src, uint dest, uint len) internal pure {
+        // Copy word-length chunks while possible
+        for (; len >= WORD_SIZE; len -= WORD_SIZE) {
+            assembly {
+                mstore(dest, mload(src))
+            }
+            dest += WORD_SIZE;
+            src += WORD_SIZE;
+        }
+
+        // Copy remaining bytes
+        uint mask = 256 ** (WORD_SIZE - len) - 1;
+        assembly {
+            let srcpart := and(mload(src), not(mask))
+            let destpart := and(mload(dest), mask)
+            mstore(dest, or(destpart, srcpart))
+        }
+    }
+
+    
+    // Creates a 'bytes memory' variable from the memory address 'addr', with the
+    // length 'len'. The function will allocate new memory for the bytes array, and
+    // the 'len bytes starting at 'addr' will be copied into that new memory.
+    function toBytes(uint256 addr, uint256 len) internal pure returns (bytes memory bts) {
+        bts = new bytes(len);
+        uint256 btsptr;
+        assembly {
```

Comment:
Can we just use `dataPtr()` here? i.e. `uint256 btsptr = dataPtr(bts)`

Reply:
Yes, we can. I'll fix that.